detectable/src/test/java/com/synopsys/integration/detectable/detectables/packagist/functional/PackagistDetectableTest.java

```diff
@@ -0,0 +1,113 @@
+ * under the License.
+ */
+package com.synopsys.integration.detectable.detectables.packagist.functional;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.Assertions;
+
+import com.synopsys.integration.bdio.model.Forge;
+import com.synopsys.integration.detectable.Detectable;
+import com.synopsys.integration.detectable.DetectableEnvironment;
+import com.synopsys.integration.detectable.Extraction;
+import com.synopsys.integration.detectable.detectables.packagist.ComposerLockDetectableOptions;
+import com.synopsys.integration.detectable.functional.DetectableFunctionalTest;
+import com.synopsys.integration.detectable.util.graph.NameVersionGraphAssert;
+
+public class PackagistDetectableTest extends DetectableFunctionalTest {
+
+    public PackagistDetectableTest() throws IOException {
+        super("composer");
+    }
+
+    @Override
+    protected void setup() throws IOException {
+        addFile(Paths.get("composer.lock"),
+            "{",
+            "   \"_readme\": [\"dummy README\"],",
+            "   \"hash\": \"3bedbf455f54168900e467a64bbe2b86\",",
+            "   \"packages\": [",
+            "       {",
+            "           \"name\": \"clue/graph\",",
+            "           \"version\": \"v0.9.0\",",
+            "           \"require\": {",
+            "               \"php\": \">=5.3.0\"",
+            "           },",
+            "           \"require-dev\": {",
+            "               \"phpunit/phpunit\": \"~4.0\"",
+            "           }",
+            "       },",
+            "       {",
+            "           \"name\": \"graphp/algorithms\",",
+            "           \"version\": \"v0.8.1\",",
+            "           \"require\": {",
+            "               \"clue/graph\": \"~0.9.0|~0.8.0\",",
+            "               \"php\": \">=5.3\"",
+            "           },",
+            "           \"require-dev\": {",
+            "               \"psr/log\": \"~1.0\",",
+            "               \"symfony/event-dispatcher\": \"~2.1\",",
+            "               \"symfony/phpunit-bridge\": \"~2.7\"",
+            "           }",
+            "       }",
+            "   ],",
+            "   \"packages-dev\": []",
+            "}"
+        );
+
+        addFile(Paths.get("composer.json"),
+            "{",
+            "   \"name\": \"clue/graph-composer\",",
+            "   \"version\": \"1.0.0\",",
+            "   \"keywords\": [\"dependency graph\", \"visualize dependencies\", \"visualize composer\"],",
+            "   \"license\": \"MIT\",",
+            "   \"require\": {",
+            "       \"php\": \"^5.3.6 || ^7.0\",",
+            "       \"clue/graph\": \"^0.9.0\",",
+            "       \"jms/composer-deps-analyzer\": \"0.1.*\",",
+            "       \"symfony/console\": \"^2.1 || ^3.0\",",
+            "       \"graphp/graphviz\": \"^0.2.0\"",
```

Comment:
I am a little concerned that the dependencies listed here don't appear in the graph. The composer.json and composer.lock came from the same project at the same time?